mgmt_cli_test.py

```diff
@@ -506,6 +512,86 @@ def test_backup_rate_limit(self):
         self.verify_backup_success(mgr_cluster=mgr_cluster, backup_task=backup_task)
         self.log.info('finishing test_backup_rate_limit')
 
+    @staticmethod
+    def _get_all_snapshot_files_s3(cluster_id, bucket_name, region_name):
+        file_set = set()
+        try:
+            s3_client = boto3.client('s3', region_name=region_name)
+            paginator = s3_client.get_paginator('list_objects')
+            pages = paginator.paginate(Bucket=bucket_name, Prefix=f'backup/sst/cluster/{cluster_id}')
+            for page in pages:
+                content_list = page["Contents"]
+                file_set.update([item["Key"] for item in content_list])
+        except KeyError as err:
+            if err.args[0] == "Contents":
+                pass  # No snapshot files of the cluster exist, probably no backup ran before this function
+            else:
+                raise
+        return file_set
+
+    @staticmethod
+    def _get_all_snapshot_files_gce(cluster_id, bucket_name):
+        file_set = set()
+        gcp_credentials = KeyStore().get_gcp_credentials()
+        gce_driver = libcloud.storage.providers.get_driver(libcloud.storage.types.Provider.GOOGLE_STORAGE)
+        driver = gce_driver(gcp_credentials["project_id"] + "@appspot.gserviceaccount.com",
+                            gcp_credentials["private_key"],
+                            project=gcp_credentials["project_id"])
+        container = driver.get_container(container_name=bucket_name)
+        dir_listing = driver.list_container_objects(container, ex_prefix=f'backup/sst/cluster/{cluster_id}')
+        for listing_object in dir_listing:
+            file_set.add(listing_object.name)
+        # Unlike S3, if no files match the prefix, no error will occur
+        return file_set
+
+    def _get_all_snapshot_files(self, cluster_id):
+        bucket_name = self.params.get('backup_bucket_location').split()[0]
+        if self.params.get('cluster_backend') == 'aws':
+            return self._get_all_snapshot_files_s3(cluster_id=cluster_id, bucket_name=bucket_name,
+                                                   region_name=self.region[0])
+        elif self.params.get('cluster_backend') == 'gce':
+            return self._get_all_snapshot_files_gce(cluster_id=cluster_id, bucket_name=bucket_name)
+        else:
+            raise ValueError(f'"{self.params.get("cluster_backend")}" not supported')
+
+    def test_backup_purge_removes_orphan_files(self):
```

Comment:
Is that  the only case of orphan files we have in the test plan?
What about decommissioned nodes files? aren't they purged as well?

Reply:
I saw that in dtest we have test for that.
So it's not under orphaned purge but under regular purge?

Reply:
Yes. This is the only test in sct because we needed to stop the backup mid-upload, something we can't do possibly in dtest.